src/Shipping/PickupLocation.php

```diff
@@ -42,6 +56,7 @@ public function calculate_shipping( $package = array() ) {
 				if ( ! $location['enabled'] ) {
 					continue;
 				}
+				$has_pickup_address = ! empty( $location['address'] ) && ! empty( $location['address']['city'] ) && ! empty( $location['address']['state'] ) && ! empty( $location['address']['country'] );
```

Comment:
Some countries like UAE don't have a state. This condition will hide the `Collection from` for them as well. 
I feel we should have helper function that will check the address completeness based on the country. 
We have one similar function: `isAddressComplete` in utils. Do you think we should utilize it here?
You can test that with the below address:
![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/46d36eae-0994-4ff8-bd2c-65ce45eac30e)